src/test/java/com/amihaiemil/eoyaml/AllYamlLinesTest.java

```diff
@@ -298,4 +335,80 @@ public void complainsWhenCannotTurn() {
             );
         }
     }
+
+    /**
+     * Unit test for serializing quoted keys of scalar values.
+     * @throws IOException When there's a problem reading the sample files.
+     */
+    @Test
+    public void quotedKeysRoundTripTests() throws IOException {
+        quotedKeysTest("quotedKeysMin.yml", 1);
+        quotedKeysTest("quotedKeysMax.yml", 6);
+    }
+
+    /**
+     * Do a round-trip test on the given quoted-key sample file.
+     * @param filename Sample file name. Contain a mapping keyed with
+     *                 'a_mapping'
+     * @param mapSize The size of the mapping
+     * @throws IOException When there's a problem reading the sample files.
+     */
+    private void quotedKeysTest(final String filename, final int mapSize)
```

Comment:
@zsandoz Please write two separate tests. It's better to have 2 atomic tests with a little duplicated code than to have this kind of private reusable tests :D

Reply:
Done